src/Microsoft.AspNetCore.Identity.Specification.Tests/IdentitySpecificationTestBase.cs

```diff
@@ -1139,6 +1206,34 @@ private static string MakeToken(string purpose, string userId)
         /// </summary>
         /// <returns>Task</returns>
         [Fact]
+        public async Task PasswordValidatorWithEmptyErrorsCanBlockResetPasswordWithStaticTokenProvider()
```

Comment:
Does this test require the FakeDataProtectionProvider?

Reply:
In SetupIdentityService, .AddDefaultTokenProviders is called, which eventually causes a DataProtectorTokenProvider to be created

```
public DataProtectorTokenProvider(IDataProtectionProvider dataProtectionProvider, IOptions<DataProtectionTokenProviderOptions> options)
```

This asks the IServiceCollection for an IDataProtectionProvider...which...was never registered, as far as I could find.  Most of the tests fail for me with the error:

```
Message: System.InvalidOperationException : Unable to resolve service for type 'Microsoft.AspNetCore.DataProtection.IDataProtectionProvider' while attempting to activate 'Microsoft.AspNetCore.Identity.DataProtectorTokenProvider`1[Microsoft.AspNetCore.Identity.EntityFrameworkCore.IdentityUser]'.
```

Adding a fake fixed all of these...though I still have many failures related to code I haven't touched.  I can definitely remove the fake if there's something else taking care of this that I'm just missing...?

Reply:
Hrm that's strange, have you tried running the tests without your changes does a clean dev branch run work on your box?